File: src/util/serverHelpers.js
var _ = require('underscore');

// for use in async.parallel()
exports.collectData = function(res, callback) {
    var body = '';

    res.on('data', function(chunk) {
        body += chunk;
    });

    res.on('error', function(err) {
        console.error('error', err);
    });

    res.on('end', function() {
        callback(null, body);
    });
};

// Given an array of objects, maps the objects to the foodTruckSchema in
// src/models/foodtruck.js
exports.convertDataToDocs = function(foodTrucksData) {
    return _.map(foodTrucksData, function(foodTruck) {
        var latitude = parseFloat(foodTruck.latitude);
        var longitude = parseFloat(foodTruck.longitude);

        if (isNaN(latitude) || isNaN(longitude)) {
            latitude = longitude = 0;
        }

        return {
            address: foodTruck.address,
            coordinates: [longitude, latitude],
            fooditems: foodTruck.fooditems,
            name: foodTruck.applicant
        };
    });
};

File: app/util/server_helpers.py
import math


def convert_data_to_docs(food_trucks_data):
    def convert(food_truck):
        # TODO(shane): think of a better way to handle missing coordinates
        latitude = float(food_truck.get('latitude', 0))
        longitude = float(food_truck.get('longitude', 0))

        if math.isnan(latitude) or math.isnan(longitude):
            latitude = longitude = 0

        return {
            'address': food_truck.get('address'),
            'fooditems': food_truck.get('fooditems'),
            'location': {
                'type': 'Point',
                'coordinates': [longitude, latitude]
            },
            'name': food_truck.get('applicant')
        }

    return map(convert, food_trucks_data)